cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1238,11 +1238,11 @@ inflate_kernel(gpu_inflate_input_s *inputs, gpu_inflate_status_s *outputs, int p
         {
             copy_stored(state, t);
         }
-        __syncthreads();
         if (state->blast)
             break;
+        __syncthreads();
     }
-
+    __syncthreads();
```

Comment:
Is this fix working on Maxwell devices?

https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#synchronization-functions
> __syncthreads() is allowed in conditional code but only if the conditional evaluates identically across the entire thread block, otherwise the code execution is likely to hang or produce unintended side effects.

If not working, try this.
```suggestion
        if (__syncthreads_and(state->blast))
            break;
    }
```